JavaScript/4-binarySearchTree.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+
+function Node(parent, data) {
+  this.data = data;
+  this.leftChild = null;
+  this.rightChild = null;
+  this.parent = parent;
+}
+
+Node.prototype.output = function() {
+  if (this.leftChild !== null) {
+    this.leftChild.output();
+  }
+  console.dir(this.data);
+  if (this.rightChild !== null) {
+    this.rightChild.output();
+  }
+};
+
+Node.prototype.search = function(key) {
+  if (this.data === key) return this;
+  if (key < this.data) {
+    if (this.leftChild === null) return null;
+    return this.leftChild.search(key);
+  } else {
+    if (this.rightChild === null) return null;
+    return this.rightChild.search(key);
+  }
+};
+
+
+Node.prototype.insert = function(data) {
+  if (data < this.data) {
+    if (this.leftChild === null) {
+      this.leftChild = new Node(this, data);
+      return;
+    }
+    this.leftChild.insert(data);
+  } else {
+    if (this.rightChild === null) {
+      this.rightChild = new Node(this, data);
+      return;
+    }
+    this.rightChild.insert(data);
+  }
+};
+
+Node.prototype.left = function() {
+  if (this.leftChild !== null) return this.leftChild.left();
+  return this;
+};
+
+function BinarySearchTree() {
+  this.root = null;
+}
+
+BinarySearchTree.prototype.output = function() {
+  if (this.root !== null) {
+    this.root.output();
+  }
+};
+
+BinarySearchTree.prototype.search = function(key) {
+  if (this.root !== null) {
+    return this.root.search(key);
+  }
+  return null;
+};
+
+BinarySearchTree.prototype.min = function() {
+  var current = null;
+  if (this.root !== null) {
+    current = this.root;
+    while (current.leftChild !== null) {
+      current = current.leftChild;
+    }
+    return current;
+  } else return null;
```

Comment:
I'd find it more readable to use braces in the `else` clause when you use them in the `if` clause generally. In this specific case you don't need `else` at all though.